codec-redis/src/test/java/io/netty/handler/codec/redis/RedisDecoderTest.java

```diff
@@ -0,0 +1,247 @@
+/*
+ * Copyright 2016 The Netty Project
+ *
+ * The Netty Project licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package io.netty.handler.codec.redis;
+
+import io.netty.buffer.ByteBuf;
+import io.netty.buffer.Unpooled;
+import io.netty.channel.embedded.EmbeddedChannel;
+import io.netty.util.IllegalReferenceCountException;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Test;
+
+import java.util.List;
+
+import static org.hamcrest.CoreMatchers.*;
+import static org.junit.Assert.*;
+
+import static io.netty.handler.codec.redis.RedisCodecTestUtil.*;
+
+/**
+ * Verifies the correct functionality of the {@link RedisDecoder}.
+ */
+public class RedisDecoderTest {
```

Comment:
You should add also some tests that verify chunking as Redis responses sometimes do not fit a TCP packet which means a response comes in multiple parts hence the decoder is called multiple times to decode one particular segment.